src/rdkafka_sasl_cyrus.c

```diff
@@ -379,17 +394,31 @@ static int rd_kafka_sasl_cyrus_cb_getsecret(sasl_conn_t *conn,
                                             int id,
                                             sasl_secret_t **psecret) {
         rd_kafka_transport_t *rktrans = context;
-        const char *password;
+        rd_kafka_broker_t *rkb        = rktrans->rktrans_rkb;
+        rd_kafka_t *rk                = rkb->rkb_rk;
+        char *password_ptr            = rk->rk_conf.sasl.password;
+
+        if (rk->rk_conf.sasl.plain_creds_cb) {
+                int size = 256;
 
-        password = rktrans->rktrans_rkb->rkb_rk->rk_conf.sasl.password;
+                password_ptr = rd_alloca(size);
+
+                if (rk->rk_conf.sasl.plain_creds_cb(rk, NULL, NULL,
+                                                    password_ptr, &size) < 0) {
+                        rd_rkb_log(rkb, LOG_ERR, "SASLPLAIN",
+                                   "SASL password does not fit in "
+                                   "128-byte receiving buffer");
```

Comment:
bug/misleading: The implementation has 256 bytes. 

Also, maybe say something like "SASL password does not fit in 256-byte receiving buffer; needs %i", size)
(This is, unless of course we didn't fail for some other reason, in which case the whole buf length story is misleading)
These later stories apply to the username above as well.